### PR TITLE
LKE-4206 Support double quotes

### PR DESCRIPTION
**Pull request by renato-rg**
Jira: https://linkurious.atlassian.net/browse/LKE-4206

### CODE REVIEW

File: src/backend/graphItemService.ts [hunk 1]
```diff
@@ -148,154 +170,89 @@ export class GraphItemService {
...
-    headers: string[];
+  public static createNodeBatches(parsedCSV: ParsedCSV): {
+    propertyKeys: string[];
     batchedRows: {indices: number[]; rows: unknown[][]; UIDs: string[]}[];
-    badRows: [RowErrorMessage, number[]][];
   } {
     const MAX_BATCH_SIZE = 10;
-    let count = 1;
-    let headers: unknown[] | undefined = undefined;
     let batchedRows: {indices: number[]; UIDs: string[]; rows: unknown[][]}[] = [];
```
Comment (francescoinfante) [inline src/backend/graphItemService.ts:199]: why not using as a type Array<{records: Array<{index: number, UID: number, rows: unknown[]}>}>?

File: src/backend/graphItemService.ts [hunk 2]
```diff
@@ -148,154 +170,89 @@ export class GraphItemService {
...
-    const tooManyOrMissingProperties: number[] = [];
-    for (const row of csv.split(/\r?\n/)) {
-      const values = row.split(',').map(GraphItemService.parseValue);
+    for (let recordIndex = 1; recordIndex <= parsedCSV.records.length; recordIndex++) {
+      const properties = parsedCSV.records[recordIndex];
```
Comment (francescoinfante) [inline src/backend/graphItemService.ts:201]: you are going one index too much with <=

**Review by francescoinfante — CHANGES_REQUESTED** [review]

File: src/backend/shared.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import {parse} from 'papaparse';
```
[the remark this answers is not in the document]
Comment (renato-rg) [reply src/backend/shared.ts:1]: @francescoinfante I didn't know how to make a shared file that is not types for 2 different builds

File: src/public/shared.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+export class CSVUtils {
+  /**
+   * Returns the parsed CSV or a global error if any.
+   * A global error can be:
+   *   "No headers provided"
+   *   "Header value is empty"
+   *   "Missing values in records: 1, 5, 20"
+   *   "Too many values in records: 1, 5, 20"
+   */
+  public static parseCSV(csvContent: string): ParsedCSV | {error: string} {
```
[the remark this answers is not in the document]
Comment (renato-rg) [reply src/public/shared.ts:18]: The flow is: (1) Select file > (2) Click on Upload to read the file > (3) Confirm all is good > (4) "Import"
We want to show the global errors in the step 2

File: src/backend/utils.ts [hunk 5]
```diff
@@ -89,10 +86,12 @@ export class GroupedErrors extends Map<RowErrorMessage, number[]> {
   }
 }
 
-export function respond(asyncHandler: (req: Request) => Promise<{[k: string]: unknown} | void>) {
+export function respond(
+  handler: (req: Request) => {[k: string]: unknown} | void | Promise<{[k: string]: unknown} | void>
+) {
   return async (req: Request, res: _Response): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (renato-rg) [reply src/backend/utils.ts:92]: why? I need to await to get the body: `const body = await ...`

File: src/backend/graphItemService.ts (hunk 1 above)
Comment (renato-rg) [reply src/backend/graphItemService.ts:199]: I agree that other types like the one you suggest are easier to build and read, but this type `batchedRows: {indices: number[]; UIDs: string[]; rows: unknown[][]}[]` is easier to use, like in the following code:
```js
    for (const batch of batchedRows) {
      i += batch.rows.length;
      try {
        if (isEdge) {
          // 3.a. Build and run a query to create edges
          const query = GraphItemService.buildEdgesQuery(params.itemType, propertyKeys, batch.rows);
          await GraphItemService.runCypherQuery(rc, query, params.sourceKey);
        } else {
          // 3.a. Build and run a query to create nodes
          const query = GraphItemService.buildNodesQuery(params.itemType, propertyKeys, batch.rows);
          const response = await GraphItemService.runCypherQuery(rc, query, params.sourceKey);
          GraphItemService.cacheGraphNodeIDs(params.itemType, response, batch.UIDs);
        }
      } catch (e) {
        log('Batch has failed', e);
        errors.add(e, batch.indices);
      }
      this.updateImportProgress(totalRecords, i);
    }
 ```
having UIDs, rows and indices separated makes it easy to separate concerns in `errors.add()`, `cacheGraphNodeIDs()` and `buildNodesQuery()`

**Pull request merged**